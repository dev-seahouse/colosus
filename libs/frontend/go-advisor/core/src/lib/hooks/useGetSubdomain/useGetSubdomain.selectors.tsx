import { useQuery } from '@tanstack/react-query';
import { useCallback } from 'react';

import { getSubdomainQuery } from './useGetSubdomain';
import type { GetSubdomainData } from './useGetSubdomain';

/**
 * returns the status of the subdomain registration
 */
export const useSelectHasRegisteredSubdomainQuery = () => {
  return useQuery({
    ...getSubdomainQuery(),
    select: useCallback((data: GetSubdomainData) => {
      return data.subdomain !== null;
    }, []),
  });
};

/**
 * returns current tenant subdomain and tradename
 */
export const useSelectSubdomainDataQuery = () => {
  return useQuery({
    ...getSubdomainQuery(),
    select: useCallback((data: GetSubdomainData) => {
      return data;
    }, []),
  });
};
