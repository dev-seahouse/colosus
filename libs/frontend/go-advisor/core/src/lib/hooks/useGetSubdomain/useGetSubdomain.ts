import { useQuery } from '@tanstack/react-query';
import { ConnectAdvisorBrandAndSubdomainApi } from '@bambu/api-client';
import type { ConnectAdvisorGetBrandAndSubdomainResponseDto } from '@bambu/api-client';
import type { QueryClient } from '@tanstack/react-query';

export type GetSubdomainData = ConnectAdvisorGetBrandAndSubdomainResponseDto;

export const getSubdomainQuery = () => ({
  queryKey: ['getSubdomain'],
  queryFn: async () => {
    const connectAdvisorBrandAndSubdomainApi =
      new ConnectAdvisorBrandAndSubdomainApi();
    const res = await connectAdvisorBrandAndSubdomainApi.getSubdomain();

    return res.data;
  },
});

export const getSubdomainLoader =
  (queryClient: QueryClient) => async (): Promise<GetSubdomainData> => {
    const query = getSubdomainQuery();

    // ⬇️ return data or fetch it
    return (
      queryClient.getQueryData(query.queryKey) ??
      (await queryClient.fetchQuery(query))
    );
  };

export interface UseGetSubdomainOptions {
  initialData?: Partial<GetSubdomainData>;
}
export function useGetSubdomain({ initialData }: UseGetSubdomainOptions = {}) {
  return useQuery({
    ...getSubdomainQuery(),
    initialData,
  });
}

export default useGetSubdomain;
