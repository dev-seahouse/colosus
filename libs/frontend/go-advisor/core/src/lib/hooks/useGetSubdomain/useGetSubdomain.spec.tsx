import {
  renderHook,
  queryClientWrapper,
  waitFor,
} from '@bambu/react-test-utils';

import useGetSubdomain from './useGetSubdomain';

describe('useGetTenantBranding', () => {
  it('should render successfully', async () => {
    const { result } = renderHook(() => useGetSubdomain(), {
      wrapper: queryClientWrapper,
    });

    await waitFor(() => expect(result.current.isSuccess).toBe(true));

    expect(result.current.data).toMatchSnapshot();
  });
});
