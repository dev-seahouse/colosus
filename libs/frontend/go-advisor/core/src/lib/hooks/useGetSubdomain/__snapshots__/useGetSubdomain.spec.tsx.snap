// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGetTenantBranding > should render successfully 1`] = `
{
  "subdomain": "",
  "tradeName": "",
}
`;
